PublishParasoftResults/ParaReportPublishService.ts

```diff
@@ -849,4 +871,83 @@ export class ParaReportPublishService {
 
         return uuid.v5(violType + ruleId + msg + severity + lineHash + uri + order, namespace);
     }
+
+    private getReferenceSarifReports = async (): Promise<FileEntry[]> => {
+        const allBuildsForCurrentPipeline: Build[] = await this.buildClient.getBuildsForSpecificPipeline(this.projectName, this.definitionId);
+        let fileEntries: FileEntry[] = [];
+        if (!this.referenceBuild) {
+            tl.debug("No reference build has been set; using the last successful build with static analysis results");
+            fileEntries = await this.buildClient.getDefaultBuildReports(allBuildsForCurrentPipeline, this.projectName, this.SARIF_ARTIFACT_NAME, FileSuffixEnum.SARIF_SUFFIX);
+        } else {
+            const specificReferenceBuilds = allBuildsForCurrentPipeline.filter(build => {
+                return build.buildNumber == this.referenceBuild;
+            });
+
+            // Check for the specific reference build exist in current pipeline
+            if (specificReferenceBuilds.length == 1) {
+                const specificReferenceBuild = specificReferenceBuilds[0];
+
+                // Check for the succeeded or paratially-succeeded results exist in the specific reference build
+                if (specificReferenceBuild.result == BuildResult.Succeeded || specificReferenceBuild.result == BuildResult.PartiallySucceeded) {
+                    let specificReferenceBuildId: number = Number(specificReferenceBuild.id);
+                    // Check for Parasoft results exist in the specific reference build
+                    const artifact: BuildArtifact = await this.buildClient.getBuildArtifact(this.projectName, specificReferenceBuildId, this.SARIF_ARTIFACT_NAME);
+                    if (artifact) {
+                        fileEntries = await this.buildClient.getBuildReportsWithId(artifact, specificReferenceBuildId, FileSuffixEnum.SARIF_SUFFIX);
+                        tl.debug(`Retrieved static analysis results from the reference build '${this.referenceBuild}'`);
+                    } else {
+                        tl.warning(`No static analysis results were found in the specified reference build: '${this.referenceBuild}'`);
+                    }
+                } else {
+                    tl.warning(`The specified reference build '${this.referenceBuild}' is not successful or unstable`);
```

Comment:
How about: The specified reference build '${this.referenceBuild}' cannot be used. Only successful or unstable builds are valid references.